Navigate to file with test failures from the console output

**Proposed behavior**
When a scenario fails, after a run inside VS Code. 
The user can click on failure stack trace to navigate to the file and line of failure.

**Acceptance criteria**
- [ ] If the failure message is
```
 ## IDE Plugins
        
        Failed Step: Display the IDE plugins
        Specification: specs/example.spec:37
        Error Message: AssertionError: false == true
        Stacktrace: 
        AssertionError: false == true
            at step (tests/step_implementation.js:67:12)
            at process._tickCallback (internal/process/next_tick.js:109:7)
```
`specs/example.spec:37` and `tests/step_implementation.js:67:12` must be links that navigate to the line that cause the test failure.

--- a/src/execution/outputChannel.ts
+++ b/src/execution/outputChannel.ts
@@ -3,6 +3,7 @@
 import * as vscode from 'vscode';
 import { LineBuffer } from './lineBuffer';
 import * as cp from 'child_process';
+import * as path from 'path';
 
 export class OutputChannel {
 	private process: cp.ChildProcess;
@@ -28,6 +29,13 @@ export class OutputChannel {
 	}
 
 	public appendOutBuf(line: string) {
+		let prefixes :string[] = ["Specification: ","at Object.<anonymous> ("]
+		for (var i = 0; i < prefixes.length; i++) {
+			var prefix = prefixes[i];
+			if(line.includes(prefix)){
+				line = line.replace(prefix, prefix + vscode.workspace.rootPath + path.sep)
+			}
+		}
 		this.outBuf.append(line);
 	}
 

--- a/test/execution/outputChannel.test.ts
+++ b/test/execution/outputChannel.test.ts
@@ -8,7 +8,7 @@ suite('Output Channel', () => {
 		name: string;
 		text: string;
 		append(value: string): void {
-			throw new Error("Method not implemented.");
+			this.text += value;
 		}
 		appendLine(value: string): void {
 			this.text = value;
@@ -18,15 +18,9 @@ suite('Output Channel', () => {
 		}
 		show(preserveFocus?: boolean): void;
 		show(column?: vscode.ViewColumn, preserveFocus?: boolean): void;
-		show(column?: any, preserveFocus?: any) {
-			throw new Error("Method not implemented.");
-		}
-		hide(): void {
-			throw new Error("Method not implemented.");
-		}
-		dispose(): void {
-			throw new Error("Method not implemented.");
-		}
+		show(column?: any, preserveFocus?: any) {};
+		hide(): void {};
+		dispose(): void {};
 
 		read() : string {
 			return this.text;

--- a/test/execution/outputChannel.test.ts
+++ b/test/execution/outputChannel.test.ts
@@ -14,7 +14,7 @@ suite('Output Channel', () => {
 			this.text = value;
 		}
 		clear(): void {
-			throw new Error("Method not implemented.");
+			this.text='';
 		}
 		show(preserveFocus?: boolean): void;
 		show(column?: vscode.ViewColumn, preserveFocus?: boolean): void;

--- a/test/execution/outputChannel.test.ts
+++ b/test/execution/outputChannel.test.ts
@@ -7,9 +7,7 @@ suite('Output Channel', () => {
 	class MockVSOutputChannel implements vscode.OutputChannel{
 		name: string;
 		text: string;
-		append(value: string): void {
-			this.text += value;
-		}
+		append(value: string): void {}
 		appendLine(value: string): void {
 			this.text = value;
 		}
@@ -30,18 +28,18 @@ suite('Output Channel', () => {
 	test('should convert specification path from relative to absolute', (done) => {
 		var chan = new MockVSOutputChannel();
 		var outputChannel = new OutputChannel(chan, "");
-		outputChannel.appendOutBuf("      Specification: specs" + path.sep + "example.spec:19");
+		outputChannel.appendOutBuf("      Specification: " + path.join("specs", "example.spec:19") + "\n");
 
-		assert.equal(chan.read(), "      Specification: "+ vscode.workspace.rootPath + "specs" + path.sep + "example.spec:19");
+		assert.equal(chan.read(), "      Specification: "+ path.join(vscode.workspace.rootPath, "specs", "example.spec:19"));
 		done();
 	});
 
 	test('should convert implementation path from relative to absolute', (done) => {
 		var chan = new MockVSOutputChannel();
 		var outputChannel = new OutputChannel(chan, "");
-		outputChannel.appendOutBuf("      at Object.<anonymous> (tests" + path.sep + "step_implementation.js:24:10)");
+		outputChannel.appendOutBuf("      at Object.<anonymous> (" + path.join("tests", "step_implementation.js:24:10") +")\n");
 
-		assert.equal(chan.read(), "      at Object.<anonymous> (tests"+ vscode.workspace.rootPath + "specs" + path.sep + "step_implementation.js:24:10)");
+		assert.equal(chan.read(), "      at Object.<anonymous> ("+ path.join(vscode.workspace.rootPath,"tests", "step_implementation.js:24:10)"));
 		done();
 	});
 });

--- a/test/execution/outputChannel.test.ts
+++ b/test/execution/outputChannel.test.ts
@@ -1,0 +1,53 @@
+import * as assert from 'assert';
+import * as path from 'path';
+import { OutputChannel } from '../../src/execution/outputChannel';
+import * as vscode from 'vscode';
+
+suite('Output Channel', () => {
+	class MockVSOutputChannel implements vscode.OutputChannel{
+		name: string;
+		text: string;
+		append(value: string): void {
+			throw new Error("Method not implemented.");
+		}
+		appendLine(value: string): void {
+			this.text = value;
+		}
+		clear(): void {
+			throw new Error("Method not implemented.");
+		}
+		show(preserveFocus?: boolean): void;
+		show(column?: vscode.ViewColumn, preserveFocus?: boolean): void;
+		show(column?: any, preserveFocus?: any) {
+			throw new Error("Method not implemented.");
+		}
+		hide(): void {
+			throw new Error("Method not implemented.");
+		}
+		dispose(): void {
+			throw new Error("Method not implemented.");
+		}
+
+		read() : string {
+			return this.text;
+		}
+	}
+
+	test('should convert specification path from relative to absolute', (done) => {
+		var chan = new MockVSOutputChannel();
+		var outputChannel = new OutputChannel(chan, "");
+		outputChannel.appendOutBuf("      Specification: specs" + path.sep + "example.spec:19");
+
+		assert.equal(chan.read(), "      Specification: "+ vscode.workspace.rootPath + "specs" + path.sep + "example.spec:19");
+		done();
+	});
+
+	test('should convert implementation path from relative to absolute', (done) => {
+		var chan = new MockVSOutputChannel();
+		var outputChannel = new OutputChannel(chan, "");
+		outputChannel.appendOutBuf("      at Object.<anonymous> (tests" + path.sep + "step_implementation.js:24:10)");
+
+		assert.equal(chan.read(), "      at Object.<anonymous> (tests"+ vscode.workspace.rootPath + "specs" + path.sep + "step_implementation.js:24:10)");
+		done();
+	});
+});

